framework/src/modules/chain/chain.js

```diff
@@ -122,15 +126,11 @@ module.exports = class Chain {
 			this.logger.debug('Initiating storage...');
 			const storage = createStorageComponent(storageConfig, dbLogger);
 
-			if (!this.options.genesisBlock) {
-				throw Error('Failed to assign nethash from genesis block');
-			}
-
 			// TODO: For socket cluster child process, should be removed with refactoring of network module
 			this.options.loggerConfig = loggerConfig;
 
 			const self = this;
-			const scope = {
+			self.scope = {
```

Comment:
why instead of deleting `self`, use self here?

Reply:
I had to move `self.scope = scope` line to top in order to have a reference to scope variable to check if `createSocketCluster` is calling the right variable. I could use `this` instead but deleting `self` doesn't seem possible because `Sequence` instance is using chain instance in the onWarning method.